crates/muvm/src/bin/muvm.rs

```diff
@@ -21,26 +22,27 @@ use muvm::launch::{launch_or_lock, LaunchResult, DYNAMIC_PORT_RANGE};
 use muvm::monitor::spawn_monitor;
 use muvm::net::{connect_to_passt, start_passt};
 use muvm::types::MiB;
+use muvm::utils::launch::{GuestConfiguration, Launch};
 use nix::sys::sysinfo::sysinfo;
 use nix::unistd::User;
 use rustix::io::Errno;
 use rustix::process::{
     geteuid, getgid, getrlimit, getuid, sched_setaffinity, setrlimit, CpuSet, Resource,
 };
-use serde::{Deserialize, Serialize};
+use serde::Serialize;
 use tempfile::NamedTempFile;
 
-#[derive(Serialize, Deserialize)]
-pub struct KrunConfig {
+#[derive(Serialize)]
+pub struct KrunConfig<'a, 'b> {
     #[serde(rename = "Cmd")]
-    args: Vec<String>,
+    args: &'a [&'b str],
```

Comment:
I would leave this as `Vec<String>`. Lifetimes are kind of a premature optimization anti-pattern here, since they can make things more confusing. There's no issue copying a few strings around here.